draft-birkholz-scitt-architecture.md

```diff
@@ -544,6 +544,29 @@ Editor's Note: This may be moved to appendix.
 
 ## Messages
 
+All messages are sent as HTTP GET or POST requests.
+
+All errors are returned with HTTP 4xx or 5xx status code as JSON objects following the OData JSON (ISO/IEC 20802-2:2016) schema below:
+
+~~~
+{
+  "error": {
+    "code": "<code>",
+    "message": "<message>"
+  }
+}
+~~~
```

Comment:
Question: why defining a custom format when you can use the standardised "Problem Details" envelope -- which is available for both [CoAP](https://datatracker.ietf.org/doc/rfc9290/) and [HTTP](https://datatracker.ietf.org/doc/draft-ietf-httpapi-rfc7807bis/) ?

Reply:
I'm not sure why you say custom. The format I used is standardised in ISO/IEC 20802-2:2016, see the sentence above.

Reply:
it's preferable to depend on open standards rather than standards behind paywalls

Reply:
Sure, I agree. The OData JSON Format is openly standardised under OASIS, see https://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#_Toc38457793, and has been republished in ISO, behind a paywall. I proposed this because it's simple (e.g., does not require minting URIs for error codes) and is already widely in use (e.g., all of Microsoft is using it). Happy to continue the discussion though! :)

Reply:
cool, maybe change the ref to OASIS.

FWIW, RFC7807 too is quite widely used (see https://github.com/search?q=RFC+7807), also IETF dogfooding and whatnot :-)

Reply:
Alright, in combination with https://datatracker.ietf.org/doc/rfc7807/referencedby/ I think it's worth following the lead of CT 2.0 https://www.rfc-editor.org/rfc/rfc9162.html#name-log-client-messages and use urns for error types:
```json
    {
        "type": "urn:ietf:params:scitt:error:invalidInput",
        "detail": "..."
    }
```

Reply:
Done, please check.

Reply:
thanks